drgn/helpers/common/stack.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Stack
+-----
+
+The ``drgn.helpers.common.stack`` module provides helpers for working with the
+stack trace.
+"""
+
+from drgn import FaultError, PlatformFlags, StackTrace
+from drgn.helpers.common.memory import identify_address
+
+__all__ = ("print_annotated_stack",)
+
+
+def print_annotated_stack(trace: StackTrace) -> None:
+    """
+    Print the content of the stack trace, identifying symbols and
+    slab objects.
+
+    :param trace: Stack trace to print.
+    """
+    prog = trace.prog
+    # platform must be known if there is a stack trace
+    assert prog.platform is not None
+    is_64_bit = prog.platform.flags & PlatformFlags.IS_64_BIT
+    word_size = 8 if is_64_bit else 4
+    hex_format = "{:016x}" if is_64_bit else "{:08x}"
+
+    if prog.platform.flags & PlatformFlags.IS_LITTLE_ENDIAN:
+        byteorder = "little"
+    else:
+        byteorder = "big"
+
+    print("STACK POINTER     VALUE")
```

Comment:
I didn't love the extra spaces for 32-bit when I looked at it. What do you think of this:

```
STACK     VALUE
POINTER
01234567: 012345678
01234567: 012345678
```

I.e., wrap the "STACK POINTER" heading so that it fits within the natural width. Something like:

```suggestion
    if prog.platform.flags & PlatformFlags.IS_64_BIT:
        print("STACK POINTER     VALUE")
    else:
        print("STACK     VALUE\nPOINTER")
```

You should consolidate this with the other `IS_64_BIT` checks if you can.

Reply:
Looks good to me - I don't have a strong preference on this, so I'll defer to you 😅

Reply:
Fixed!